src/minifig_detector.py

```diff
@@ -0,0 +1,196 @@
+"""minifig向き検出クラス.
+
+@file minifig_detector.py
+@author Hara1274
+"""
+import cv2
+import numpy as np
+import onnxruntime as ort
+import os
+import json
+
+
+class MinifigDetector:
+    """minifigの向きを検出するクラス."""
+
+    def __init__(self):
+        """検出器を初期化."""
+        self.session = None
+        self.input_name = None
+        self.labels = ["front", "back", "right", "left"]
+        self.conf_threshold = 0.25
+        self.nms_threshold = 0.45
+        self.input_size = 640
+
+        # YOLO11 ONNXモデルのパス
+        project_root = os.path.dirname(os.path.dirname(__file__))
+        model_path = os.path.join(
+            project_root, "models", "11s_100epoch_&_650imgsz_fig.onnx")
+
+        # モデルファイルが存在する場合のみonnxモデルの設定
+        if os.path.exists(model_path):
+            # 推論セッションを作成
+            self.session = ort.InferenceSession(model_path)
+            #  ONNXモデルの入力層の名前を取得
+            self.input_name = self.session.get_inputs()[0].name
+
+    def preprocess_image(self, img):
+        """推論用に画像を前処理.
+
+        Args:
+            img: 入力画像
+
+        Returns:
+            tuple: (処理後画像, スケール比, パディング情報)
+        """
+        # アスペクト比を保持するスケール計算
+        shape = img.shape[:2]  # 元画像サイズ (H, W)
+        # アスペクト比維持のスケール
+        r = min(self.input_size / shape[0], self.input_size / shape[1])
+        # スケール後サイズ (W, H)
+        new_unpad = (int(round(shape[1] * r)), int(round(shape[0] * r)))
+        img_resized = cv2.resize(
+            img, new_unpad, interpolation=cv2.INTER_LINEAR)
+
+        # 640x640にするための余白計算
+        dw = self.input_size - new_unpad[0]  # 水平余白
+        dh = self.input_size - new_unpad[1]  # 垂直余白
+        top, bottom = dh // 2, dh - dh // 2  # 上下分割
+        left, right = dw // 2, dw - dw // 2  # 左右分割
+
+        # 灰色（114）でパディングして640x640に調整
+        img_padded = cv2.copyMakeBorder(
+            img_resized, top, bottom, left, right, cv2.BORDER_CONSTANT,
+            value=(114, 114, 114))
+
+        # YOLO入力形式に変換: HWC→CHW, [0,1]正規化, バッチ次元追加
+        img_input = img_padded.transpose(2, 0, 1).astype(
+            np.float32) / 255.0  # (3,640,640)
+        img_input = np.expand_dims(img_input, axis=0)  # (1,3,640,640)
+
+        return img_input, r, (left, top)
+
+    def postprocess(self, pred, scale, pad):
+        """推論結果を後処理.
+
+        Args:
+            pred: 推論出力 [1, 8, N] (YOLO11形式)
+            scale: レターボックスのスケール比
+            pad: パディング情報 (left, top)
+
+        Returns:
+            str: 検出された向き
```

Comment:
だいたい辞書型が返ってる気がするんだけど、これでおけ？

```suggestion
        Returns:
            dict: 検出結果 {"wasDetected": bool, "direction": str,
                         "confidence": float}
```